data_gateway/configuration.py

```diff
@@ -1,188 +1,241 @@
+}
 
-    :param float mics_freq: microphones sampling frequency
-    :param float mics_bm: TODO
-    :param float baros_freq: barometers sampling frequency
-    :param float baros_bm: TODO
+DEFAULT_SLEEP_STATES = {"0": "Exiting sleep", "1": "Entering sleep"}
+
+DEFAULT_INFO_TYPES = {"0": "Battery info"}
+
+DEFAULT_SAMPLES_PER_PACKET = {
+    "Mics": 8,
+    "Diff_Baros": 24,
+    "Baros_P": 1,
+    "Baros_T": 1,
+    "Acc": 40,  # IMU, int(240 / 2 / 3)
+    "Gyro": 40,  # IMU, int(240 / 2 / 3)
+    "Mag": 40,  # IMU, int(240 / 2 / 3)
+    "Analog Vbat": 60,
+    "Constat": 24,
+}
+
+
+DEFAULT_SENSOR_CONVERSION_CONSTANTS = {
+    "Mics": 1,
+    "Diff_Baros": 1,
+    "Baros_P": 40.96,
+    "Baros_T": 100,
+    "Acc": 1,
+    "Gyro": 1,
+    "Mag": 1,
+    "Analog Vbat": 1,
+    "Constat": 1,
+}
+
+DEFAULT_SENSOR_COMMANDS = {
+    "start": ["startBaros", "startDiffBaros", "startIMU", "startMics"],
+    "stop": ["stopBaros", "stopDiffBaros", "stopIMU", "stopMics"],
+    "configuration": ["configBaros", "configAccel", "configGyro", "configMics"],
+    "utilities": [
+        "getBattery",
+        "setConnInterval",
+        "tpcBoostIncrease",
+        "tpcBoostDecrease",
+        "tpcBoostHeapMemThr1",
+        "tpcBoostHeapMemThr2",
+        "tpcBoostHeapMemThr4",
+    ],
+}
+
+DEFAULT_NUMBER_OF_SENSORS = {
+    "Mics": 10,
+    "Baros_P": 40,
+    "Baros_T": 40,
+    "Diff_Baros": 5,
+    "Acc": 3,
+    "Gyro": 3,
+    "Mag": 3,
+    "Analog Vbat": 1,
+    "Constat": 4,
+}
+
+DEFAULT_SESSION = {
+    "label": None,
+}
+
+
+class GatewayConfiguration:
+    """A data class containing configured/default valuer for the gateway receiver
+
+    :param float baudrate: serial port baud rate
+    :param Literal["little", "big"] endian: one of "little" or "big"
+    :param string installation_reference: A unique reference (id) for the current installation
+    :param float latitude: The latitude of the turbine in WGS84 coordinate system
+    :param float longitude: The longitude of the turbine in WGS84 coordinate system
+    :param int packet_key_offset: The value from which each node's packet key is calculated (packet_key = node_id + packet_key_offset)
+    :param string receiver_firmware_version: The version ofthe firmware running on the gateway receiver, if known.
+    :param int serial_buffer_rx_size: serial receiving buffer size in bytes
+    :param int serial_buffer_tx_size: serial transmitting buffer size in bytes
+    :param string turbine_id: A unique id for the turbine on which this is installed
+    :return None:
+    """
+
+    def __init__(
+        self,
+        baudrate=2300000,
+        endian="little",
+        installation_reference="unknown",
+        latitude=0,
+        longitude=0,
+        packet_key_offset=245,
+        receiver_firmware_version="unknown",
+        serial_buffer_rx_size=4095,
+        serial_buffer_tx_size=1280,
+        turbine_id="unknown",
+    ):
+        self.baudrate = baudrate
+        self.endian = endian
```

Comment:
Shall we add some validation that either "little" or "big" is given?

Reply:
No, it'll be unrecognised in the serial port reader and barf instantly if incorrect.